[server] Fetch GitLab repositories for a given user or group

jjmerchante opened the pull request:
This PR retrieves all the repositories from a given user or a given group and its subgroups.
As in GitHub, it returns for each repository the URL, whether it is a fork or not, and if it has issues enabled.
There is a redis queue job to fetch the repositories from GitLab API and it can be started from the GraphQL API.
Users can provide a token for the GitLab backend and it will use it.

This PR also includes fixtures for loading supported backends (GitHub and GitLab):
```
python manage.py loaddata --settings=<settings file> backends.json
```

Closes: https://github.com/chaoss/grimoirelab-bestiary/issues/110

jjmerchante committed:
[fixtures] Create fixtures for GitHub and GitLab backend

This file contains the list of backends supported by Bestiary.
It can be imported using the next command:

python manage.py loaddata --settings <filename> backends.json

The contents will be inserted in the "datasourcetype" table.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

jjmerchante committed:
[GLOwners] Include a function to retrieve repositories from an GitLab owner

This commit fetches from GitLab API all the repositories from a specific
user or group and its subgroups. Include information related like the url,
whether it has issues enabled or not and if it is a fork.
It includes the related tests.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

jjmerchante committed:
[job] Include a job to fetch GitLab repositories from an owner

This commit includes the implementation of a redis queue job
to fetch all the repositories from a specific owner and store
the result or the errors in the output.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

jjmerchante committed:
[schema] Include a graphql query and mutation to fetch GitLab repositories

This commit includes the implementation for graphql to fetch GitHub
repositories from a given user or group in a job.
It returns information like the url, whether it has issues
enabled or not and if it is a fork.
Include related tests.

Signed-off-by: JJMerchante <jj.merchante@gmail.com>

evamillan commented:
@jjmerchante It's a corner case I've only found while testing it from the UI, but if you try to fetch the repositories with an invalid token it returns a "GitLab owner "[owner]" not found" error, so I guess it's failing to catch the actual error somewhere.

jjmerchante commented:
> @jjmerchante It's a corner case I've only found while testing it from the UI, but if you try to fetch the repositories with an invalid token it returns a "GitLab owner "[owner]" not found" error, so I guess it's failing to catch the actual error somewhere.

I rebased and updated the branch with those changes fixed.

evamillan reviewed (approved):
LGTM
